governance/default.go

```diff
@@ -536,16 +536,13 @@ func (g *Governance) ParseVoteValue(gVote *GovernanceVote) (*GovernanceVote, err
 		val = string(v)
 	case params.GoverningNode:
 		v, ok := gVote.Value.([]uint8)
-		if !ok || len(v) != common.AddressLength {
```

Comment:
Because this is a string, the length should be 42. 
20 bytes in hex string = 40 characters
'0x' = 2 characters.

Shouldn't we need to check this as 42 instead of 20?
Or, do we have another validation logic after address translation?

Reply:
@kjhman21 @aidan-kwon 
I checked, and 20 byte is correct length.

Assumed askii value is inserted, the original address may be "0xf8c9c61c5e7f2b6219d1c28b94e5cb3cdc802594".

Its `[]uint8` representation is "307866386339633631633565376632623632313964316332386239346535636233636463383032353934",
and you can see "6332386239346535636233636463383032353934" at the end 20 bytes.

I tested with v1.7.0-rc.1/v1.7.0-rc.2 binary to check whether the vote api can accept 42 bytes of not, and it did not accept 42 byte.
```
> governance.vote("governance.addvalidator", "307866386339633631633565376632623632313964316332386239346535636233636463383032353934")
Error: Your vote couldn't be placed. Please check your vote's key and value
    at web3.js:3276:20
    at web3.js:6450:15
    at web3.js:5214:36
    at <anonymous>:1:1

> governance.vote("governance.addvalidator", "0x307866386339633631633565376632623632313964316332386239346535636233636463383032353934")
Error: Your vote couldn't be placed. Please check your vote's key and value
    at web3.js:3276:20
    at web3.js:6450:15
    at web3.js:5214:36
    at <anonymous>:1:1

> governance.vote("governance.addvalidator", 307866386339633631633565376632623632313964316332386239346535636233636463383032353934)
Error: Your vote couldn't be placed. Please check your vote's key and value
    at web3.js:3276:20
    at web3.js:6450:15
    at web3.js:5214:36
    at <anonymous>:1:1

> governance.vote("governance.addvalidator", 0x307866386339633631633565376632623632313964316332386239346535636233636463383032353934)
Error: Your vote couldn't be placed. Please check your vote's key and value
    at web3.js:3276:20
    at web3.js:6450:15
    at web3.js:5214:36
    at <anonymous>:1:1
```

On the other hand, when voting  ("governance.addvalidator", "0xf8c9c61c5e7f2b6219d1c28b94e5cb3cdc802594") at private network using v1.7.0-rc.2/v1.8.0-rc.1, the gVote.Value in `ParseVoteValue` is "0xf8c9c61c5e7f2b6219d1c28b94e5cb3cdc802594".

```
func (g *Governance) ParseVoteValue(gVote *GovernanceVote) (*GovernanceVote, error) {
	var val interface{}
	k := GovernanceKeyMap[gVote.Key]

	// filter out if vote value is an interface list
	if reflect.TypeOf(gVote.Value) == reflect.TypeOf([]interface{}{}) {
		return nil, ErrValueTypeMismatch
	}

	switch k {
	case params.GovernanceMode, params.MintingAmount, params.MinimumStake, params.Ratio:
		val = string(gVote.Value.([]uint8))
	case params.GoverningNode, params.AddValidator, params.RemoveValidator:
		val = common.BytesToAddress(gVote.Value.([]uint8)) // here!
```

Reply:
@yoomee1313  `BytesToAddress` doesn't reject invalid length addresses as you can the comment in the below.
And, we used various client versions, v1.5 ~ v1.7, at the incident time to figure out the root cause of the issue in a short time.
```
// BytesToAddress returns Address with value b.
// If b is larger than len(h), b will be cropped from the left.
```

Reply:
Understood.

Reply:
@aidan-kwon thanks for letting me know about it. I'll try with various client versions.